repair_order_group/models/repair_order.py

```diff
@@ -0,0 +1,243 @@
+
+        Features:
+        • Consolidated SO creation for repair groups
+        • Standard individual SO creation for non-grouped repairs
+        • Mixed selection handling with focused UX
+        • Detailed validation messages with repair order references
+
+        Returns:
+            dict: Action to view the created sale order(s)
+        """
+
+        # Split selected repairs into grouped and ungrouped
+        grouped = self.filtered(lambda r: r.group_id)
+        ungrouped = self - grouped
+
+        # No groups at all → fallback to core logic
+        if not grouped:
+            return super().action_create_sale_order()
+
+        # Core validations (same as original Odoo, but applied once)
+        concerned = self.filtered("sale_order_id")
+        if concerned:
+            ref_str = "\n".join(concerned.mapped("name"))
+            raise UserError(
+                _(
+                    "You cannot create a quotation for a repair order that is "
+                    "already linked to an existing sale order.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        no_partner = self.filtered(lambda ro: not ro.partner_id)
+        if no_partner:
+            ref_str = "\n".join(no_partner.mapped("name"))
+            raise UserError(
+                _(
+                    "You need to define a customer for a repair order in order to "
+                    "create an associated quotation.\n"
+                    "Concerned repair order(s):\n%(ref_str)s",
+                    ref_str=ref_str,
+                )
+            )
+
+        so_vals_list = []
+
+        # One Sale Order per unique repair group
+        unique_groups = grouped.mapped("group_id")
+        for _group in unique_groups:
+            valid_repairs = _group.repair_ids._get_valid_repairs_for_so()
+
+            if not valid_repairs:
+                continue
+
+            # Use first valid repair for common SO fields
+            first = valid_repairs[0]
+
+            so_vals_list.append(
+                {
+                    "company_id": first.company_id.id,
+                    "partner_id": first.partner_id.id,
+                    "warehouse_id": (
+                        first.picking_type_id.warehouse_id.id
+                        if first.picking_type_id and first.picking_type_id.warehouse_id
+                        else False
+                    ),
+                    "repair_order_ids": [(4, repair.id) for repair in valid_repairs],
+                }
+            )
+
+        # Bulk create all SOs for grouped repairs
+        if so_vals_list:
+            sale_orders = self.env["sale.order"].create(so_vals_list)
+
+            # Single call for ALL move lines creation
+            all_so_repairs = sale_orders.mapped("repair_order_ids")
+            all_so_repairs.move_ids._create_repair_sale_order_line()
```

Comment:
No need to to use "mapped" here. Since Odoo 16.0  dot-notation for for x2many fields works the same way. So simply
```python
sale_orders.repair_order_ids.move_ids._create_repair_sale_order_line()
```